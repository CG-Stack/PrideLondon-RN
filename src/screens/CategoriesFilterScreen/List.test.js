// @flow
import React from "react";
import { shallow } from "enzyme";
import List from "./List";

const render = props => shallow(<List {...props} />);

describe("List Component", () => {
  it("renders correctly", () => {
    const stagedCategories: Set<string> = new Set(["Music"]);
    const onPress = () => {};
    const component = render({ locale: "en-GB", stagedCategories, onPress });

    expect(component).toMatchSnapshot();
  });
});
