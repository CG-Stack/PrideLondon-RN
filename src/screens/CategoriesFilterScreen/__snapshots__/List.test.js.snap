// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component renders correctly 1`] = `
<ShadowedScrollView
  style={
    Object {
      "backgroundColor": "rgb(45, 47, 127)",
    }
  }
>
  <ListItem
    category={
      Object {
        "color": "#ff5b44",
        "label": "Cabaret and Variety",
      }
    }
    key="Cabaret and Variety"
    onPress={[Function]}
    selected={false}
  />
  <ListItem
    category={
      Object {
        "color": "#2acfff",
        "label": "Community",
      }
    }
    key="Community"
    onPress={[Function]}
    selected={false}
  />
  <ListItem
    category={
      Object {
        "color": "#2c77e5",
        "label": "Talks and Debates",
      }
    }
    key="Talks and Debates"
    onPress={[Function]}
    selected={false}
  />
  <ListItem
    category={
      Object {
        "color": "#7de2d1",
        "label": "Film and Screenings",
      }
    }
    key="Film and Screenings"
    onPress={[Function]}
    selected={false}
  />
  <ListItem
    category={
      Object {
        "color": "#fc4984",
        "label": "Plays and Theatre",
      }
    }
    key="Plays and Theatre"
    onPress={[Function]}
    selected={false}
  />
  <ListItem
    category={
      Object {
        "color": "#00a3bd",
        "label": "Social and Networking",
      }
    }
    key="Social and Networking"
    onPress={[Function]}
    selected={false}
  />
  <ListItem
    category={
      Object {
        "color": "#ffd95e",
        "contrast": true,
        "label": "Nightlife",
      }
    }
    key="Nightlife"
    onPress={[Function]}
    selected={false}
  />
  <ListItem
    category={
      Object {
        "color": "#bc01ff",
        "label": "Exhibition and Tours",
      }
    }
    key="Exhibition and Tours"
    onPress={[Function]}
    selected={false}
  />
  <ListItem
    category={
      Object {
        "color": "#c3d30e",
        "label": "Sports and Activities",
      }
    }
    key="Sports and Activities"
    onPress={[Function]}
    selected={false}
  />
  <ListItem
    category={
      Object {
        "color": "#ff88f1",
        "label": "Health",
      }
    }
    key="Health"
    onPress={[Function]}
    selected={false}
  />
  <ListItem
    category={
      Object {
        "color": "#5770d1",
        "label": "Music",
      }
    }
    key="Music"
    onPress={[Function]}
    selected={true}
  />
</ShadowedScrollView>
`;
